### PR TITLE
Simplify script to fetch latest CLI version from Axon

### PR DESCRIPTION
**Pull request by dexamundsen**
Changes similar to https://github.com/verily-src/terra-tool-cli/pull/342

### CODE REVIEW

**Review by yuhuyoyo — APPROVED** [review]

**Review by rogerwangcs — APPROVED** [review]

File: startupscript/vertex-ai-user-managed-notebook/post-startup.sh [hunk 1]
```diff
@@ -398,31 +398,13 @@ readonly TERRA_SERVER="$(get_metadata_value "instance/attributes/terra-cli-serve
...
-      ;;
-    verily)
-      afsservice=terra-axon.api.verily.com
-      ;;
-    *)
-      >&2 echo "ERROR: $TERRA_SERVER is not a known verily server."
+  # Map the CLI server to appropriate AFS service path and fetch the CLI distribution path
+  versionJson="$(curl -s "https://${TERRA_SERVER/verily/terra}-axon.api.verily.com/version")" || exit_code=$?
+  if [[ $exit_code -ne 0 ]]; then
+      >&2 echo "ERROR: $TERRA_SERVER is not a known server"
```
Comment (mbookman) [inline startupscript/vertex-ai-user-managed-notebook/post-startup.sh:404]:
```
     >&2 echo "ERROR: ${TERRA_SERVER} is not a known server"
```

**Review by mbookman — CHANGES_REQUESTED** [review]
Some small nits

File: startupscript/dataproc/startup.sh [hunk 2]
```diff
@@ -364,31 +364,13 @@ readonly TERRA_SERVER="$(get_metadata_value "instance/attributes/terra-cli-serve
...
-      ;;
-    verily)
-      afsservice=terra-axon.api.verily.com
-      ;;
-    *)
-      >&2 echo "ERROR: $TERRA_SERVER is not a known verily server."
+  # Map the CLI server to appropriate AFS service path and fetch the CLI distribution path
+  versionJson="$(curl -s "https://${TERRA_SERVER/verily/terra}-axon.api.verily.com/version")" || exit_code=$?
+  if [[ $exit_code -ne 0 ]]; then
+      >&2 echo "ERROR: $TERRA_SERVER is not a known server"
```
[the remark this answers is not in the document]
Comment (dexamundsen) [reply startupscript/dataproc/startup.sh:370]: done

File: startupscript/dataproc/startup.sh [hunk 3]
```diff
@@ -364,31 +364,13 @@ readonly TERRA_SERVER="$(get_metadata_value "instance/attributes/terra-cli-serve
...
       exit 1
-      ;;
-  esac
-  # Build AFS service path and fetch the CLI distribution path
-  versionJson="$(curl -s "https://$afsservice/version")"
-  cliDistributionPath=$(echo "$versionJson" | jq -r '.cliDistributionPath')
+  fi
+  cliDistributionPath="$(echo "$versionJson" | jq -r '.cliDistributionPath')"
```
[the remark this answers is not in the document]
Comment (dexamundsen) [reply startupscript/dataproc/startup.sh:373]: done

File: startupscript/vertex-ai-user-managed-notebook/post-startup.sh (hunk 1 above)
Comment (dexamundsen) [reply startupscript/vertex-ai-user-managed-notebook/post-startup.sh:404]: done

File: startupscript/dataproc/startup.sh [hunk 4]
```diff
@@ -364,31 +364,13 @@ readonly TERRA_SERVER="$(get_metadata_value "instance/attributes/terra-cli-serve
...
 
   ${RUN_AS_LOGIN_USER} "\
     curl -L https://storage.googleapis.com/${cliDistributionPath#gs://}/download-install.sh | TERRA_CLI_SERVER=${TERRA_SERVER} bash && \
```
[the remark this answers is not in the document]
Comment (dexamundsen) [reply startupscript/dataproc/startup.sh:376]: This is part of the multiline string already within quotes

File: startupscript/vertex-ai-user-managed-notebook/post-startup.sh [hunk 5]
```diff
@@ -793,7 +776,6 @@ fi
 # Restart proxy to pick up the new env
 ######################################
 readonly APP_PROXY=$(get_metadata_value "instance/attributes/terra-app-proxy")
-readonly PROXY_ENV="/opt/deeplearning/proxy.env"
```
Comment (dexamundsen) [inline startupscript/vertex-ai-user-managed-notebook/post-startup.sh:796]: not used

**Comment by dexamundsen** [comment]
Added some cleanup, hence re-requested review

File: startupscript/post-startup.sh [hunk 6]
```diff
@@ -97,34 +99,52 @@ ${RUN_AS_USER} "mv '${JAVA_DIRNAME}' '${USER_HOME_LOCAL_SHARE}'"
...
- curl -L https://storage.googleapis.com/workbench-public/workbench-cli/download-install.sh | bash"
+# Fetch the VWB CLI server environment from the metadata server to install appropriate CLI version
+TERRA_SERVER="$(get_metadata_value "instance/attributes/terra-cli-server")"
+if [[ -n "${TERRA_SERVER}" ]]; then
+  TERRA_SERVER="verily"
+fi
 
-cp terra ${TERRA_INSTALL_PATH}
+# If the server environment is a verily server, use the verily download script.
+if [[ $TERRA_SERVER == *"verily"* ]]; then
```
Comment (mbookman) [inline startupscript/post-startup.sh:118]:
```
if [[ "${TERRA_SERVER}" == *"verily"* ]]; then
```

File: startupscript/post-startup.sh [hunk 7]
```diff
@@ -97,34 +99,52 @@ ${RUN_AS_USER} "mv '${JAVA_DIRNAME}' '${USER_HOME_LOCAL_SHARE}'"
...
+  # Map the CLI server to appropriate AFS service path and fetch the CLI distribution path
+  versionJson="$(curl -s "https://${TERRA_SERVER/verily/terra}-axon.api.verily.com/version")" || exit_code=$?
+  if [[ $exit_code -ne 0 ]]; then
```
Comment (mbookman) [inline startupscript/post-startup.sh:121]:
```
 if [[ "${exit_code}" -ne 0 ]]; then
```

File: startupscript/post-startup.sh [hunk 8]
```diff
@@ -201,16 +223,22 @@ ${RUN_AS_USER} "cd '${TERRA_GIT_REPOS_DIR}' && terra git clone --all"
...
+# Installs gcsfuse if it is not already installed.
+if ! which gcsfuse >/dev/null 2>&1; then
+  emit "Installing gcsfuse..."
+  # install packages needed to install gcsfuse
+  apt-get install -y \
+    fuse \
+    lsb-core
+
+  # Install based on gcloud docs here https://cloud.google.com/storage/docs/gcsfuse-install.
+  export GCSFUSE_REPO=gcsfuse-$(lsb_release -c -s) \
```
Comment (mbookman) [inline startupscript/post-startup.sh:235]:
```
  export GCSFUSE_REPO="gcsfuse-$(lsb_release -c -s)" \
```

File: startupscript/vertex-ai-user-managed-notebook/post-startup.sh [hunk 9]
```diff
@@ -389,48 +389,31 @@ ${RUN_AS_LOGIN_USER} "\
...
-emit "Installing the Terra CLI ..."
+# Install & configure the VWB CLI
+emit "Installing the VWB CLI ..."
 
 # Fetch the Terra CLI server environment from the metadata server to install appropriate CLI version
-readonly TERRA_SERVER="$(get_metadata_value "instance/attributes/terra-cli-server")"
+TERRA_SERVER="$(get_metadata_value "instance/attributes/terra-cli-server")"
+if [[ -n "${TERRA_SERVER}" ]]; then
+  TERRA_SERVER="verily"
+fi
```
Comment (mbookman) [inline startupscript/vertex-ai-user-managed-notebook/post-startup.sh:399]: As above, mark the variable readonly.

File: startupscript/vertex-ai-user-managed-notebook/post-startup.sh [hunk 10]
```diff
@@ -389,48 +389,31 @@ ${RUN_AS_LOGIN_USER} "\
...
-      afsservice=terra-preprod-axon.api.verily.com
-      ;;
-    verily)
-      afsservice=terra-axon.api.verily.com
-      ;;
-    *)
-      >&2 echo "ERROR: $TERRA_SERVER is not a known verily server."
+  # Map the CLI server to appropriate AFS service path and fetch the CLI distribution path
+  versionJson="$(curl -s "https://${TERRA_SERVER/verily/terra}-axon.api.verily.com/version")" || exit_code=$?
+  if [[ $exit_code -ne 0 ]]; then
```
Comment (mbookman) [inline startupscript/vertex-ai-user-managed-notebook/post-startup.sh:405]:
```
  if [[ "${exit_code}" -ne 0 ]]; then
```

**Review by mbookman — CHANGES_REQUESTED** [review]
Just a few small things...

File: startupscript/dataproc/startup.sh [hunk 11]
```diff
@@ -359,56 +357,38 @@ ${RUN_AS_LOGIN_USER} "\
 emit "Installing the VWB CLI ..."
 
 # Fetch the VWB CLI server environment from the metadata server to install appropriate CLI version
-readonly TERRA_SERVER="$(get_metadata_value "instance/attributes/terra-cli-server")"
+TERRA_SERVER="$(get_metadata_value "instance/attributes/terra-cli-server")"
+if [[ -n "${TERRA_SERVER}" ]]; then
+  TERRA_SERVER="verily"
+fi
```
[the remark this answers is not in the document]
Comment (dexamundsen) [reply startupscript/dataproc/startup.sh:365]: done, in all files

File: startupscript/post-startup.sh (hunk 8 above)
Comment (dexamundsen) [reply startupscript/post-startup.sh]: done, in all files

File: startupscript/vertex-ai-user-managed-notebook/post-startup.sh (hunk 10 above)
Comment (dexamundsen) [reply startupscript/vertex-ai-user-managed-notebook/post-startup.sh]: missed it, done in all files

**Review by mbookman — APPROVED** [review]
LGTM!

**Pull request merged**